File: code/ch01/1.3.3.dailyTemperatures.java
class Solution {
  public int[] dailyTemperatures(int[] T) {
    Stack<Integer> stack = new Stack<Integer>();
    int[] ans = new int[T.length];
    for (int i = 0; i < T.length; i++) {
      while (!stack.empty() && T[i] > T[stack.peek()]) {
        int top = stack.pop();
        ans[top] = i - top;
      }
      stack.push(i);
    }
    return ans;
  }
}

File: code/ch01/1.3.3. dailyTemperatures.cpp
class Solution {
public:
  vector<int> dailyTemperatures(vector<int> &T) {
    stack<int> st;
    vector<int> ans(T.size(), 0);
    for (int i = 0; i < T.size(); i++) {
      while (!st.empty() && T[i] > T[st.top()]) {
        int top = st.top();
        st.pop();
        ans[top] = i - top;
      }
      st.push(i);
    }
    return ans;
  }
};